diff --git a/auto_entitylabel.module b/auto_entitylabel.module
index 2bf156b..1b202c1 100644
--- a/auto_entitylabel.module
+++ b/auto_entitylabel.module
@@ -66,6 +66,34 @@ function auto_entitylabel_inline_entity_form_entity_form_alter(&$form, &$form_st
   auto_entitylabel_prepare_entityform($form, $entity);
 }
 
+/**
+ * Implements hook_inline_entity_form_table_fields_alter().
+ */
+function auto_entitylabel_inline_entity_form_table_fields_alter(&$fields, $context) {
+  // Replace label field behavior.
+  if (isset($fields['label'])) {
+    $fields['label']['type'] = 'callback';
+    $fields['label']['callback'] = 'auto_entitylabel_inline_entity_label_callback';
+  }
+}
+
+/**
+ * A callback function to provide autoentitylabel for inline entity form.
+ */
+function auto_entitylabel_inline_entity_label_callback($entity, $variables) {
+  $autolabel = $entity->label();
+  if ($entity instanceof ContentEntityInterface) {
+    $decorator = \Drupal::service('auto_entitylabel.entity_decorator');
+    /** @var \Drupal\auto_entitylabel\AutoEntityLabelManager $decorated_entity */
+    $decorated_entity = $decorator->decorate($entity);
+    if ($decorated_entity->hasLabel() && $decorated_entity->autoLabelNeeded()) {
+      $autolabel = $decorated_entity->setLabel();
+    }
+
+  }
+  return $autolabel;
+}
+
 /**
  * Prepares the label replacement in the entity form.
